### PR TITLE
#9620: Update studio to support variable components (Mailer)

### PR DESCRIPTION
**Pull request by raph90**
Reference to #9620. 

Currently Studio supports `export function Welcome()` syntax for emails. This PR adds support for `export const Welcome = () => ...` 

Also added optional dependencies for SWC to fix an error that _I think_ is architecture related.

### CODE REVIEW

**Comment by Tobbe** [comment]
@raph90 Thanks for your PR. Looks like you need to run `yarn install` (and possibly follow that up with `yarn check`) and commit the updated yarn.lock file as well.

**Comment by raph90** [comment]
> @raph90 Thanks for your PR. Looks like you need to run `yarn install` (and possibly follow that up with `yarn check`) and commit the updated yarn.lock file as well.

Thanks @Tobbe, this is now ready for review I think.

File: packages/studio/api/mail/index.ts
Comment (Tobbe) [inline packages/studio/api/mail/index.ts:220]:
```suggestion
function extractParamAndValue(component: any): {
```

File: packages/studio/api/mail/index.ts
Comment (Tobbe) [inline packages/studio/api/mail/index.ts:267]: Is this comment still correct?

**Comment by raph90** [comment]
Updated based on your comments @Tobbe

**Comment by Tobbe** [comment]
@Josh-Walker-GM I'll leave the final review to you

File: packages/studio/api/mail/index.ts
Comment (Josh-Walker-GM) [inline packages/studio/api/mail/index.ts:272]: It looks like we only handle the `ExportDeclaration` and `ExportDefaultExpression` in the `extractParamAndValue` function. Including `VariableDeclaration` here also means that any non-exported variable is getting picked up and treated as a default export. 

I think we should remove those two types from here and only support them for just now without worrying about the `ExportDefaultDeclaration` as that is even more complex.

**Review by Josh-Walker-GM — CHANGES_REQUESTED** [review]
I think there a small thing to correct in the AST processing which I've commented explicitly about. Otherwise I'm happy with this and we can get it merge once that's addressed. Thanks!

File: packages/studio/api/mail/index.ts
Comment (raph90) [reply packages/studio/api/mail/index.ts:272]: Ah ok, thanks @Josh-Walker-GM , I'll have a look into that

**Review by Josh-Walker-GM — APPROVED** [review]
@raph90 I pushed up some changes - I hope you don't mind. By changing the way swc was imported it now is typed so that made things easier to work with.

Would you be able to test this out and confirm everything is now working as you'd expect? After that I'll merge.

**Comment by raph90** [comment]
All looking good to me @Josh-Walker-GM! 

Weirdly I am still getting the 
```
Error: Fallback bindings does not support filesystem access
    at Compiler.parseFileSync 
``` 
error, but when I actually go into studio and look at the templates everything is working. It might be because the project I'm testing on is running Redwood 6.0.7 rather than canary, I'm not sure. But I think this is good to go if you're happy!

**Pull request merged**
